### ISSUE TITLE
[Feature Request] Dataset Support

### ISSUE THREAD

**Issue by rohanshah** [sbg/sevenbridges-python#104]
Hi, I wasn't sure the best place to make this request or ask about functionality but...

Is there planned support for querying datasets and files in datasets?

I saw there are "advanced access" endpoints in the API for querying this way but couldn't find any documentation or anything in the code that suggested that the python client supported this yet.

Thanks!

**Comment by damirkrstanovic** [comment]
Hi @rohanshah  - are you referring to the datasets like TCGA on the CGC ? There is a HAL based query API which we didn't wrap into sevenbridges-python. 

As a workaround, you can utilize something like [HALEasy](https://pypi.org/project/HALEasy/) to talk to this API. The only thing to note is that you need to specify CGC authentication.
Something like this should do the trick:
```
from haleasy import HALEasy
from requests import Session
mysession = Session()
mysession.headers.update({'X-SBG-Auth-Token':'<TOKEN HERE>'})
h = HALEasy('https://cgc-datasets-api.sbgenomics.com', session=mysession)
```
After this you can utilize standard HAL functionality. Note that other HAL libraries should also work - more resources on https://github.com/mikekelly/hal_specification

**Comment by rohanshah** [comment]
Hey @damirkrstanovic I was referring more to the [datasets API](http://docs.cavatica.org/docs/cavatica-datasets-api-overview) (I think it's in the advanced API right now). Essentially something similar to the `api.project` module.

Right now I'm using the underlying `api = sbg.Api(...)` object to make requests directly against the `/files` endpoints with `dataset=...` query param (instead of `project=...`) in order to get the basic file information for the dataset. Then I'm using the `api.files` module to actually request a `File` object (with metadata).

So basically just an extension of the `Api` object to include `dataset`.

**Comment by damirkrstanovic** [comment]
Thanks for clarification @rohanshah. Currently we support datasets as described only on the CAVATICA platform - I assume you are interested in working on this platform ?

**Comment by damirkrstanovic** [comment]
Thanks @rohanshah - should not be hard to do and your request makes sense. Will report when we schedule this enhancement.

**Comment by rohanshah** [comment]
@damirkrstanovic awesome! thank you!

**Comment by QuicksilverMachine** [comment]
Added in 0.12.1 and documented [here](http://sevenbridges-python.readthedocs.io/en/latest/quickstart/#managing-datasets).
